### PR TITLE
Added a way to invoke external functions.

### PR DESCRIPTION
**Pull request by EgorDm**
The plugin currently does not support functions that are not defined within the current serverless.yml.
This commit adds a way to define an endpoint for these functions so the can be called anyway as a rest call.
One can make a function callable by defining events.http section.

### CODE REVIEW

**Comment by bboure** [comment]
@EgorDm Thank you for your contrib.
Could you please elaborate the use case for this?

Is this supposed to work with a secondary stack and the `serverless-offline` plugin?
Could you provide a full example, including the dataSource ?

Thanks,

**Comment by EgorDm** [comment]
Hey @bboure,
Sorry for a late response. 
I have just added an [example](https://github.com/bboure/serverless-appsync-simulator/blob/9ac7fabcbe390ca07aedc1857599a6c8cb2a571c/examples/external-functions/serverless.yml) on how it should be used.

This is indeed supposed to be used with a second stack. For example, if a lambda function is defined in a separate serverless.yml from the appsync config. 
When using serverless offline, one can run it using an HTTP request (for example http://localhost:3016/2015-03-31/functions/myLambda/invocations).
When deploying one can usually specify an ARN for a function to be called. But while offline it is not possible to invoke it.

File: examples/external-functions/serverless.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+              field: getPosts
+              request: Query.getPosts.request.vtl
+              response: Query.getPosts.response.vtl
+        dataSources:
+            - type: AWS_LAMBDA
+              name: MyLambda
+              config:
+                  functionName: myLambda
+                  lambdaFunctionArn: "arn:aws:lambda:${self:provider.region}:*:getPosts-graphql"
+        functions:
```
Comment (bboure) [inline examples/external-functions/serverless.yml:29]: @EgorDm Please move this into `custom.appsync-simulator` since this is simulator related.

**Review by bboure — CHANGES_REQUESTED** [review]
Thank you @EgorDm 
I made a few comments. Apart from that, LGTM!

File: examples/external-functions/serverless.yml (hunk 1 above)
Comment (EgorDm) [reply examples/external-functions/serverless.yml]: I have moved the function config into the plugin (serverless-appsync-simulator) config.
It is indeed useful to add this separation.

File: src/getAppSyncConfig.js [hunk 2]
```diff
@@ -60,6 +61,23 @@ export default function getAppSyncConfig(context, appSyncConfig) {
...
+          const func = conf.functions[functionName];
+          return {
+            ...dataSource,
+            invoke: async (payload) => {
+              const result = await axios.request({
+                url: func.url,
+                method: func.method,
+                data: payload,
+              });
+              return result.data;
```
[the remark this answers is not in the document]
Comment (EgorDm) [reply src/getAppSyncConfig.js:76]: This is indeed a tricky one. I have tested if it is indeed necessary.

When a server responds with an API error (for example a 404). It should still have a json body. In this case, axios still returns a valid response where the json body should still be handled by graphql. (tested by chaning the url to https://jsonplaceholder.typicode.com/postz).

When an unrecoverable error occurs, for example network error. An exception is indeed thrown. It seems though that it is correctly handled by appsync simulator itself.

[the remark this answers is not in the document]
Comment (bboure) [reply src/getAppSyncConfig.js:76]: @EgorDm sounds good.
A possible alternative would be to set the `validateStatus` parameter to false.

File: README.md [hunk 3]
```diff
@@ -202,6 +203,20 @@ For now, the supported resources to be automatically resovled by `Ref:` are:
...
+
+```yaml
+custom:
+  appSync:
+    functions:
+      addUser:
+        url: http://localhost:3016/2015-03-31/functions/addUser/invocations
+      addPost:
+        url: https://jsonplaceholder.typicode.com/posts
+```
```
Comment (bboure) [inline README.md:217]: @EgorDm you missed updating this.

Comment (EgorDm) [reply README.md:217]: Thanks, this should be fixed now.

File: src/getAppSyncConfig.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (EgorDm) [reply src/getAppSyncConfig.js:77]: I have added validateStatus: false such that no error is thrown on invalid status codes. The lambda resolver should handle it.
Handling of other is done by appsync simulator itself (similar to [invoke](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-nodejs-function-runtime-provider/src/utils/invoke.ts#L56))

**Review by bboure — APPROVED** [review]
Awesome. Thanks!!!

**Pull request merged**